lib/apiservers/engine/backends/system.go

```diff
@@ -14,90 +14,62 @@
 
 package vicbackends
 
+//****
+// system.go
+//
+// Rules for code to be in here:
+// 1. No remote or swagger calls.  Move those code to system_portlayer.go
+// 2. Always return docker engine-api compatible errors.
+//		- Do NOT return fmt.Errorf()
+//		- Do NOT return errors.New()
+//		- DO USE the aliased docker error package 'derr'
+//		- It is OK to return errors returned from functions in system_portlayer.go
+
 import (
-	"bufio"
 	"fmt"
-	"net/http"
-	"os"
-	"regexp"
 	"runtime"
-	"strings"
 	"time"
 
 	"golang.org/x/net/context"
 
 	log "github.com/Sirupsen/logrus"
 
 	"github.com/vmware/vic/lib/apiservers/engine/backends/cache"
-	"github.com/vmware/vic/lib/apiservers/portlayer/client"
-	"github.com/vmware/vic/lib/apiservers/portlayer/client/containers"
-	"github.com/vmware/vic/lib/apiservers/portlayer/client/misc"
 	"github.com/vmware/vic/pkg/trace"
 
-	derr "github.com/docker/docker/errors"
-	"github.com/docker/docker/pkg/parsers/kernel"
+	//	derr "github.com/docker/docker/errors"
 	"github.com/docker/docker/pkg/platform"
-	"github.com/docker/docker/pkg/system"
 	"github.com/docker/engine-api/types"
 	"github.com/docker/engine-api/types/events"
 	"github.com/docker/engine-api/types/filters"
+	"github.com/docker/go-units"
 )
 
 type System struct {
 }
 
-type ContainerStatus struct {
-	count      int
-	numRunning int
-	numStopped int
-	numPaused  int
-}
-
 const (
-	etcReleaseFile    = "/etc/os-release"
-	usrlibReleaseFile = "/usr/lib/os-release"
-	unknown           = "<unknown"
+	systemStatusMhz    = " VCH mhz limit"
+	systemStatusMemory = " VCH memory limit"
+	systemOS           = " VMware OS"
+	systemOSVersion    = " VMware OS version"
+	systemProductName  = " VMware Product"
 )
 
-func (s *System) SystemInfo() (*types.Info, error) {
-	defer trace.End(trace.Begin("SystemInfo"))
-
-	// Use docker pkgs to get some system data
-	kernelVersion := unknown
-	if kv, err := kernel.GetKernelVersion(); err != nil {
-		log.Warnf("Could not get kernel version: %v", err)
-	} else {
-		kernelVersion = kv.String()
-	}
-
-	operatingSystem := getOperatingSystem()
-
-	meminfo, err := system.ReadMemInfo()
-	if err != nil {
-		log.Errorf("Could not read system memory info: %v", err)
-	}
-
-	// Check if portlayer server is up
-	plClient := PortLayerClient()
+// Override this for unit testing
```

Comment:
Probably worth a comment for what this is when *not* unit testing vs when unit testing.